ivre/db/mongo.py

```diff
@@ -1287,6 +1294,38 @@ def convert(val):
             update["$set"]["traces"] = doc['traces']
         return update
 
+    @staticmethod
+    def migrate_schema_hosts_11_12(doc):
+        """Converts a record from version 11 to version 12. Version 12 changes
+the structured output for fcrdns and rpcinfo script.
+
+        """
+        assert doc["schema_version"] == 11
+        update = {"$set": {"schema_version": 12}}
+        updated = False
+        for port in doc.get('ports', []):
+            for script in port.get('scripts', []):
+                if script['id'] == "fcrdns":
+                    if "fcrdns" in script:
+                        # Since dots in keys are converted to
+                        # underscores, we need to convert them
+                        # back. It's OK to do so because underscores
+                        # cannot be used in FQDNs.
+                        script["fcrdns"] = [
```

Comment:
You might want to use your function `change_fcrdns`.

Reply:
`change_fcrdns()` does not replace back `_` by `.`.